sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1426,6 +1428,12 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         Minute(expression(ctx.source))
       case "SECOND" =>
         Second(expression(ctx.source))
+      case "MILLISECONDS" | "MSEC" =>
```

Comment:
Ur, @MaxGekk . Did you check PostgreSQL to find out all supported **variants**?
Please check the variants really instead of relying on my comments. 😅

Reply:
Let me be clear. I didn't check PostgreSQL full variants, but, at least, `MS` and `MSECS` are missing here.

cc @maropu . Could you help @MaxGekk to complete this feature?

Reply:
> Did you check PostgreSQL to find out all supported variants?

I checked PostgreSQL documentation and JIRA ticket that mention `milliseconds` only. As far as I see `timestamp.sql` uses `msec` / `msec` as well.

> at least, MS and MSECS are missing here.

Do we need to support them if they are not used in tests?

Reply:
> Please check the variants really instead of relying on my comments. 😅

I relied on existing test https://github.com/apache/spark/blob/caa23e3efd3f4422e8f599b30bec3ef1fb33c03c/sql/core/src/test/resources/sql-tests/inputs/pgSQL/timestamp.sql#L198-L199 not on your comment, sorry :-E

Reply:
I've checked the related code in pgsql and it seems to have many variants:
- millisecon
- ms
- msec
- mseconds
- msecs
- msecond

The implementation of `extract` is `timestamp_part` in pgsql: https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/timestamp.c#L4507
This function handles `DTK_MILLISEC ` for milleseconds: https://github.com/postgres/postgres/blob/fded4773eb60541c6e7dbcf09c9bcb1cd36a063b/src/backend/utils/adt/timestamp.c#L4552
`DTK_MILLISEC ` is expanded into many variants above:
https://github.com/postgres/postgres/blob/fded4773eb60541c6e7dbcf09c9bcb1cd36a063b/src/interfaces/ecpg/pgtypeslib/dt_common.c#L446
https://github.com/postgres/postgres/blob/fded4773eb60541c6e7dbcf09c9bcb1cd36a063b/src/include/utils/datetime.h#L49